src/interfaces.d.ts

```diff
@@ -1,10 +1,27 @@
+interface IAppProps {
+
+}
+
+interface IAppState {
+  showJamModal: boolean;
+}
+
 interface IJamInputProps {
   defaultMessage: string;
-  onSubmit: (jamString: string) => void
+  onSubmit: (jamString: string, jamResult: string) => void
 }
 
 interface IJamInputState {
   jamString: string;
   jamText: string;
 }
 
+interface IJamModalProps {
+  show: boolean;
+  imgSrc: string;
+  close: () => void;
+}
+
+interface IJamModalState {
+
+}
```

Comment:
Seems a little strange to put all the interfaces in one file. Is that the recommended way to do it? Or is it just how we're doing it for now and then if/when it makes sense to break them up we can?

Reply:
I used a boilerplate to app that had the interfaces defined like that, but whether that is the correct way or not I have not fully investigated. It may be time to break those out into separate interface files.